### ISSUE TITLE
Configurable product not showing price on category and product page

### ISSUE THREAD

**Issue by hemendraajmera** [magento/magento2#28068]
### Preconditions (*)

1. using Magento 2.3.5-p1
2. added new configurable product with two product

### Steps to reproduce (*)

1. after adding configurable products with two products and provide stock and status as in stock. on the category page, it showing price and on product detail page it showing option and price
![image](https://user-images.githubusercontent.com/2478037/80807318-7a58e900-8bda-11ea-934a-708f96cec5c3.png)
![image](https://user-images.githubusercontent.com/2478037/80807360-a5433d00-8bda-11ea-96dd-f01bdbef69a8.png)

2. chaining status to out of stock for it not showing product price on the category page and on product detail page also not showing price and option 
![image](https://user-images.githubusercontent.com/2478037/80807208-36fe7a80-8bda-11ea-9e17-286c9a602af9.png)
![image](https://user-images.githubusercontent.com/2478037/80807389-b3915900-8bda-11ea-98b6-2b47ed445dd6.png)

### Expected result (*)

1. show the price in whatever status of the product
![image](https://user-images.githubusercontent.com/2478037/80807318-7a58e900-8bda-11ea-934a-708f96cec5c3.png)
![image](https://user-images.githubusercontent.com/2478037/80807360-a5433d00-8bda-11ea-96dd-f01bdbef69a8.png)

### Actual result (*)

1. not showing price in out of stock 
![image](https://user-images.githubusercontent.com/2478037/80807208-36fe7a80-8bda-11ea-9e17-286c9a602af9.png)
![image](https://user-images.githubusercontent.com/2478037/80807389-b3915900-8bda-11ea-98b6-2b47ed445dd6.png)

**Comment by shikhamis11** [comment]
hi @hemendraajmera 
I also believe that product price should display for out of stock configurable product just like simple products.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @shikhamis11 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-33935` were created

**Issue Available**: @shikhamis11, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by hemendraajmera** [comment]
@magento give me 2.5-p1-develop instance

**Comment by magento-engcom-team** [comment]
Hi @hemendraajmera. Thank you for your request. I'm working on Magento 2.5-p1-develop instance for you

**Comment by hemendraajmera** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @hemendraajmera. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @hemendraajmera, here is your [Magento instance](https://i-28068-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-28068-2-4-develop.instances.magento-community.engineering/admin_c1cb](https://i-28068-2-4-develop.instances.magento-community.engineering/admin_c1cb)
Login: `fc82cd42` Password: `781a45e1d9a5`
Instance will be terminated in up to 3 hours.

**Comment by hemendraajmera** [comment]
the same issue with provided instance, please refer below
after set status out of stock it showing as 
![price and option not  showing](https://user-images.githubusercontent.com/2478037/80934691-ec9e1900-8de6-11ea-9fa1-068fb05d6053.png)
after set status in stock with configurable product also 
![showing price and option](https://user-images.githubusercontent.com/2478037/80934696-ef990980-8de6-11ea-9e96-1aea3db6fcfa.png)

**Comment by magento2dev** [comment]
Hello,
I can confirm it stills an issue on Magento 2.3.4

**Comment by madhurajawat** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Magento instance deployment temporary unavailable due to service maintenance.

**Comment by madhurajawat** [comment]
@magento I am working on this

**Comment by hemendraajmera** [comment]
Any quick solutions for issue Please.

**Comment by madhurajawat** [comment]
> Any quick solutions for issue Please.

@hemendraajmera Price does not come When all associated products are out of stock in Magento.
You can do code customization with overriding the Magento\Catalog\Pricing\Render\FinalPriceBox.php code to show the pricing.

Thanks

**Comment by magento2dev** [comment]
Sadly @madhurajawat you can't simply override this renderer because the ordering by prices on category pages do not work. Magento indexes for configurable products prices still equal to 0 so it's not working.

Thanks

**Comment by madhurajawat** [comment]
> Sadly @madhurajawat you can't simply override this renderer because the ordering by prices on category pages do not work. Magento indexes for configurable products prices still equal to 0 so it's not working.
> 
> Thanks

@magento2dev I have checked with this and it's working. Also working category page ordering.
With this, it will show the price of the configurable product on the product page. You can customize code to display the lowest product price as well. 
For the custom module, Check reference below: 
https://magento.stackexchange.com/questions/160145/magento-2-how-to-show-prices-for-out-of-stock-products

Thanks!

**Comment by magento2dev** [comment]
Thanks @madhurajawat but the custom module you are referring to is not for configurable products. If you apply the same fix on Magento\ConfigurableProduct\Pricing\Render\FinalPriceBox the product ordering on category pages is not working.

**Comment by madhurajawat** [comment]
> Thanks @madhurajawat but the custom module you are referring to is not for configurable products. If you apply the same fix on Magento\ConfigurableProduct\Pricing\Render\FinalPriceBox the product ordering on category pages is not working.

@magento2dev Yes, I have checked it myself. the category page is also working well at my end.

![Screenshot_2020-05-18 Jackets - Tops - Women(1)](https://user-images.githubusercontent.com/59955325/82210460-c0c57c00-992c-11ea-877e-055c337f32c7.png)
![Screenshot_2020-05-18 Jackets - Tops - Women](https://user-images.githubusercontent.com/59955325/82210465-c327d600-992c-11ea-9eab-e351ea987126.png)
![Screenshot_2020-05-18 Juno Jacket](https://user-images.githubusercontent.com/59955325/82210472-c4f19980-992c-11ea-840e-ef8b004fa337.png)

Thanks!

**Comment by magento2dev** [comment]
@madhurajawat these products are out of stock ?
because I just tried on a fresh Magento installation and it's not working, many people have the same issue. I check the catalog_product_index_price table data for out of stock configurable products and all prices are equal to 0 so it can't work.

Thanks

**Comment by madhurajawat** [comment]
> @madhurajawat these products are out of stock ?
> because I just tried on a fresh Magento installation and it's not working, many people have the same issue. I check the catalog_product_index_price table data for out of stock configurable products and all prices are equal to 0 so it can't work.
> 
> Thanks

@magento2dev Yes, It is out of stock products. You can see the out of stock label as well.

**Comment by magento2dev** [comment]
@madhurajawat on a fresh Magento 2.3.4 it's not working if configurable and sub simple products are set out of stock, the above custom module display the prices but the ordering is not working.

**Comment by magento2dev** [comment]
Thanks for your help @madhurajawat but I already looked at this tickets and there is no real fix for this issue.

**Comment by hemendraajmera** [comment]
any updates

**Comment by hemendraajmera** [comment]
@sureshp1000  Thanks for the quick solution.

I tried to implement the mentioned change on my website, it working and It showing the price as low as a section on category and product page. 
but the variation options like size, color, etc. dropdown not showing on that configurable product.

**Comment by sureshp1000** [comment]
Hi @hemendraajmera  Thanks for your response.

As you know the issue was not showing price for the configurable product so now we have price on the product page. 

Regarding the variation options like size, color, etc. drop-down not showing on that configurable product, it is Magento default feature that we are not able to add to cart the out of stock variations product. And now we have all associated products out of stock hence these options are disabled.

**Comment by nordcomputer** [comment]
I just created a small module to show the price of out-of-stock items based on [this answer](https://magento.stackexchange.com/a/298250/84488) at stackexchange.  Feel free to fork:
https://github.com/nordcomputer/magento2-showoutofstockprice

Composer install should work, if you add the repo to your dependencies. Be aware, that I set the required php version to "^7.1" - you may need to reconfigure that in the cmposer.json file, if you use another version.

**Comment by hemendraajmera** [comment]
Thank you @nordcomputer to provide the solution to show price for out of the stock product. I haven't tried to install but need to install to resolve the Magento bug.
before I install the extension, can you please help me to answer the below query:
After using extension it also shows options for configurable products which not visible for out of the stock configurable product.

**Comment by nordcomputer** [comment]
Sorry, can you rephrase your question? The extension will show "out-of-stock" options for configurable products as "greyed out" and not clickable. When there is no option for the configurable product in stock, then the options (all together) dont show up.

**Comment by Moppelfett** [comment]
Hi,
I still have this issue on a fresh 2.4.3-p1 Version.
Are there any updates?

**Comment by bselimbuyukelcibemeir** [comment]
Since someone came up with a custom module, Magento team would not give a fuck and bother themselves to fix a very important issue.

**Comment by engcom-Bravo** [comment]
Hi @hemendraajmera,

Thanks for your reporting and collaboration.

As per this https://github.com/magento/magento2/issues/16069#issuecomment-517785774 we are closing this issue.

Thanks.
